### PR TITLE
4362 redesign fugu showcase page

### LINKED ISSUE #4362: Design: Redesign /fugu-showcase page on web.dev
Provide a design for a gallery-style page like fugu-showcase:

Original webpage: https://googlechromelabs.github.io/fugu-showcase/data/

Inlined iframe: https://developer.chrome.com/blog/fugu-showcase/

Preview URL: https://deploy-preview-2887--developer-chrome-com.netlify.app/fugu-showcase

### PR DESCRIPTION
**Pull request by devnook**
Originally based on https://github.com/GoogleChrome/developer.chrome.com/pull/4818

Fixes #4362

Adds external data source in external/fugu-showcase
Exposes that data in 11ty _data
Adds a page at /en/fugu-showcase url that links to /blog/showcase

### CODE REVIEW

**Comment by devnook** [comment]
It looks like I need to get fugu-showcase to externals on main before I can merge this PR. Extracted here: https://github.com/GoogleChrome/developer.chrome.com/pull/5161

File: site/_data/fuguShowcase.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Extract fugu items and api names from fugu showcase data.
```
Comment (tomayac) [inline site/_data/fuguShowcase.js:18]:
```suggestion
 * @fileoverview Extract Fugu items and API names from Fugu showcase data.
```

Comment (tomayac) [reply site/_data/fuguShowcase.js:18]: (This may be overly nit-picky.)

File: site/_data/i18n/common.yml [hunk 2]
```diff
@@ -197,6 +197,15 @@ website:
 featured:
   en: Featured
 
+launch_app:
+  en: 'Launch app'
+
+source_code:
+  en: 'Source code'
```
Comment (tomayac) [inline site/_data/i18n/common.yml:204]: Maybe to fit the previous item "View source", so it's verb-action-driven?

File: site/_includes/partials/showcase-card.njk [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+  id="{{ item.title | slug }}"
+  data-terms="{{ (item.title | lower) + " " + (searchApis | lower) + " " + (item.appURL | lower) }}"
+  data-used-apis="{{ apis }}"
+  class="blog-card fugu-card rounded-lg pad-300 md:pad-400 width-full"
+>
+  <div class="blog-card__thumbnail gap-bottom-300">
+    <a href="{{ item.appURL }}" rel="noopener" target="_blank">
+      <picture>
+        <source media="(prefers-color-scheme: dark)" srcset="https://googlechromelabs.github.io/fugu-showcase/data/{{ item.screenshot }}">
+        <source media="(prefers-color-scheme: light)" srcset="https://googlechromelabs.github.io/fugu-showcase/data/{{ item.screenshotDark }}">
```
Comment (tomayac) [inline site/_includes/partials/showcase-card.njk:21]:
```suggestion
        <source media="(prefers-color-scheme: light)" srcset="https://googlechromelabs.github.io/fugu-showcase/data/{{ item.screenshot }}">
        <source media="(prefers-color-scheme: dark)" srcset="https://googlechromelabs.github.io/fugu-showcase/data/{{ item.screenshotDark }}">
```

File: site/_includes/partials/showcase-card.njk [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+        <img alt="Screenshot of {{ item.title }}" height="800" src="1tuner.com.webp" width="1280">
```
Comment (tomayac) [inline site/_includes/partials/showcase-card.njk:22]:
```suggestion
        <img alt="Screenshot of {{ item.title }}." height="800" src="1tuner.com.webp" width="1280">
```

**Review by tomayac — COMMENTED** [review]
Just some minor nits, apart from the AI list in chat. Thanks for working on this!

File: site/_includes/partials/showcase-card.njk [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+  data-terms="{{ id + (item.title | lower) + " " + (searchApis | lower) + " " + (item.appURL | lower) }}"
+  data-used-apis="{{ apis }}"
+  class="blog-card fugu-card rounded-lg pad-300 md:pad-400 width-full"
+>
+  <div class="blog-card__thumbnail gap-bottom-300">
+    <a href="{{ item.appURL }}" rel="noopener" target="_blank">
+      <picture>
+        <source srcset="https://googlechromelabs.github.io/fugu-showcase/data/{{ item.screenshot }}">
+        <source media="(prefers-color-scheme: dark)" srcset="https://googlechromelabs.github.io/fugu-showcase/data/{{ item.screenshotDark }}">
+        <img class="hairline " alt="Screenshot of {{ item.title }}." height="800" src="{{ item.screenshot }}" width="1280">
```
Comment (tomayac) [inline site/_includes/partials/showcase-card.njk:23]:
```suggestion
        <img class="hairline " alt="Screenshot of {{ item.title }}." height="800" src="{{ item.screenshot }}" width="1280" loading="lazy">
```

**Comment by tomayac** [comment]
I really love this. Four small additional ideas after playing with it:

- Can the search field get a little clear **x** in the input field, so it can be emptied with one click?  
- Can the search field do its job `oninput`, so you see in real-time how the results are found?
- Can you add a **Clear filters** button, so it's easier to reset the filters?
- Can clicking the `#` copy the URL to the clipboard?

**Comment by tomayac** [comment]
And yet another idea:

- Can you add a "Show all" button that resets the API filter and clears the search field? This way it's possible to go from a [deeplink](https://deploy-preview-5159--developer-chrome-com.netlify.app/fugu-showcase#adobe-creative-cloud-sign-in) back to the overall view.

**Comment by tomayac** [comment]
Spotted a problem on narrow viewports where the search field is no longer visible:

Screenshot 2023-02-06 at 17 22 22

Screenshot 2023-02-06 at 17 22 29

File: site/en/fugu-showcase/index.njk [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+      <div class="search-box__inner" role="presentation">
+        <span class="search-box__btn">
+          {{ icon('search', {hidden: false}) }}
+        </span>
+        <input
+          id="search-fugu-apps"
+          class="search-box__input"
+          placeholder="Search apps"
+          aria-label="Search apps"
+          aria-autocomplete="list"
```
Comment (matthiasrohmer) [inline site/en/fugu-showcase/index.njk:43]: Since you are `| i18n` some of the UI copy in L46ff I think it's either none or all 😄

**Review by matthiasrohmer — COMMENTED** [review]
image

Styling for multi-line select items is off.

The search input confused me. As there is no "Go" or "Search" button I assumed search on keyup.

image

- Rendered aspect ratio for screenshots does not match intrinsic aspect ratio.

image

- Thats probably something for @tomayac: some entries feel spamy, if they are falling back to the page description (?). The XBox Cloud Gaming one for example.

**Comment by devnook** [comment]
> Styling for multi-line select items is off.

Do you mean the double-line item? I think it works as per design. Or something else?

Re: falling back to the page description => I think this is the primary way how Tom's script is populating data.

**Comment by devnook** [comment]
Most of the other issues should be resolved. 

Regarding: "Show all" button that resets the API filter - there is a bug in `enhanced-select` that prevents me from implementing it - I'd need to work with John on it outside of this PR.

**Comment by tomayac** [comment]
I think this can go live in its current form. We can always refine it later. Things on my wishlist, all of which nonessential:

- Copy to clipboard when the `#` is clicked.
- Integration of the Web Share API (including app screenshot) on supporting browsers. 

Thanks so much for working on this!

**Comment by devnook** [comment]
Thanks Tom! Copying already works, altohugh maybe we need a better way of informing user that this happened. The other request is tracked here: https://github.com/GoogleChrome/developer.chrome.com/issues/5184

**Comment by tomayac** [comment]
> Thanks Tom! Copying already works, altohugh maybe we need a better way of informing user that this happened.

Oh, nice! Yes, a little toast or so would be neat.

> The other request is tracked here: #5184

Super! Thank you!

**Review by matthiasrohmer — APPROVED** [review]
Something new 🎉

**Comment by tomayac** [comment]
@devnook, we can silent-launch this, and once @rachelandrew has reviewed the [article draft](https://docs.google.com/document/d/1kW3OSn2OaQBlF_hwMeqViAHup557WKqFtFzVcZKLvxQ/edit), we can make some noise about the new URL. (As part of my upcoming PR, I will add the redirect from `/blog/fugu-showcase/` to `/fugu-showcase`.)

**Pull request merged**